nixos/modules/services/misc/sonarr.nix

```diff
@@ -52,11 +134,19 @@ in
       description = "Sonarr";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
-
+      environment = settingsEnvVars;
       serviceConfig = {
         Type = "simple";
         User = cfg.user;
         Group = cfg.group;
+        EnvironmentFile = cfg.environmentFiles;
+        LoadCredential = lib.optional (cfg.apiKeyFile != null) "SONARR__AUTH__APIKEY:${cfg.apiKeyFile}";
+        ExecStartPre = lib.optionalString (cfg.apiKeyFile != null) pkgs.writeShellScript "sonarr-apikey" ''
+          if [ ! -s config.xml ]; then
+            echo '<?xml version="1.0" encoding="UTF-8"?><Config><ApiKey></ApiKey></Config>' > "${cfg.dataDir}/config.xml"
+          fi
+          ${lib.getExe pkgs.xmlstarlet} ed -L -u "/Config/ApiKey" -v "$(${config.systemd.package}/bin/systemd-creds cat SONARR__AUTH__APIKEY)" "${cfg.dataDir}/config.xml"
```

Comment:
The above line will leak the secret in process listings.

Reply:
@bjornfor please help me out.

I have two options: set it as env-var (`SONARR__AUTH__APIKEY`) or replace it in the config file (`config.xml`). These are my two options. How would I prevent process listing?

Reply:
I would probably use xmlstarlet like above, but have it inject `@API_KEY@` into config.xml and then use `pkgs.replace-secret` to put the actual key into `@API_KEY@` inside config.xml.

(Maybe xmlstarlet can read values from files directly?)

Reply:
@bjornfor like this? 1c0d92070874ed1637e0c5c711c98f9ec973663c

Reply:
Yes, that looks good.

Reply:
thanks for your help and fast response